Salesforce CRM (destination) Initial commit

mohan-h opened the pull request:
1. Created a new folder `crm` under `help\destinations\catalog\crm` to which the _Salesforce CRM_ **destination** connector has been added. 
2. Note have not added an _overview.md_ file, kindly suggest if an overview page is required and which existing page I can leverage as a base template.

mohan-h committed:
Salesforce CRM (destination) Initial commit

Salesforce CRM (destination) Initial commit

vfalon commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+
+### Use case #1
+
+Add customer contacts within an account to track their engagement journey.
```
These use cases are rather vague. Can you please add more context to them? See the examples in the template: https://experienceleague.adobe.com/docs/experience-platform/destinations/destination-sdk/document-destination/self-service-template.html?lang=en.

vfalon commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+
+1. You need to have a Salesforce account.
+    * Go to the Salesforce [trial](https://www.salesforce.com/in/form/signup/freetrial-sales/) page to register and create a Salesforce account, if you do not have one already.
+1. After logging in to the Salesforce portal, you will need to create a connected app within Salesforce.
```
Steps 2 and 3 go into way too much detail on the Salesforce side, which can change in the future. Please update steps 2 and 3 to something like this:

2. [Create a connected app](https://help.salesforce.com/s/articleView?id=sf.connected_app_about.htm&type=5) in the Salesforce user interface, and [enable its OAuth settings](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US).
3. Select the following [scopes](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US) for the connected app in the Salesforce user interface:

```
- chatter_api
- lightning
- visualforce
- content
- openid
- full
- api
- web
- refresh_token
- offline_access
```

Also, remove the screenshots from Salesforce UI. We don't need to document their workflows.

vfalon commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+
+![Consumer Key and Secret](../../assets/catalog/crm/salesforce/connected-app-keys.png)
+
+>[!IMPORTANT]
```
Remove this note after changing steps 2 and 3.

vfalon commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+> * Ensure you select **[!UICONTROL Enable OAuth Settings]**.
+> * Select these scopes: chatter_api, lightning, visualforce, content, openid, full, api, web, refresh_token, offline_access.
+
+Before activating data to the Salesforce CRM destination, you must have a [schema](https://experienceleague.adobe.com/docs/experience-platform/xdm/schema/composition.html), a [dataset](https://experienceleague.adobe.com/docs/platform-learn/tutorials/data-ingestion/create-datasets-and-ingest-data.html?lang=en), and [segments](https://experienceleague.adobe.com/docs/platform-learn/tutorials/segment ...
```
Line 54: Move this item to the top of the prerequisites section.

vfalon commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+
+|Target Identity|Description|Considerations|
+|---|---|---|
+|email |Email address|Note both plain text and SHA256 hashed email addresses are supported by [!DNL Adobe Experience Platform]. If the Experience platform source field contains unhashed attributes, check the **[!UICONTROL Apply transformation]** option, to have [!DNL Platform] automatically hash the data on activation.<br/><br/> Note that **Salesforce CRM** does not support hashed email addresses ...
```
The note that you added contradicts the section in the template. If Salesforce only accepts plain text, please remove the first section.

vfalon commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+> 
+>To connect to the destination, you need the **[!UICONTROL Manage Destinations]** [access control permission](/help/access-control/home.md#permissions). Read the [access control overview](/help/access-control/ui/overview.md) or contact your product administrator to obtain the required permissions.
+
+To connect to this destination, follow the steps described in the [destination configuration tutorial](https://experienceleague.adobe.com/docs/experience-platform/destinations/ui/connect-destination.html). In the configure destination workflow, fill in the fields listed in the two sections below.
```
Lines 81-87: please remove all this. This is already in the destination connection tutorial.

mohan-h replied:
Believe you meant this as the content referred content is part of the [self-service-template.md](https://github.com/AdobeDocs/experience-platform.en/blob/main/help/destinations/destination-sdk/docs-framework/self-service-template.md).

```
* Within the [!DNL Experience Platform], navigate to **Destinations**. 
* Select the **Catalog** tab and search for *Salesforce CRM*, or select *CRM* within **Categories**. 
* Then select **[!UICONTROL Set up]**. After you have established a connection to the destination, the UI label changes to **[!UICONTROL Activate Segments]**.
```

vfalon commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+
+To authenticate to the destination, fill in the required fields and select **[!UICONTROL Connect to destination]**.
+
+1. Next, you are shown a wizard which assists you in configuring the Salesforce CRM destination. Create the new destination by selecting **[!UICONTROL Configure new destination]** and fill in the details below and select **[!UICONTROL Connect to destination]**.
```
Line 92: No need for this step. This is part of the destination config tutorial. Just leave the screenshot and the parameters below it.

mohan-h replied:
Believe you meant this as the content referred content is part of the [self-service-template.md](https://github.com/AdobeDocs/experience-platform.en/blob/main/help/destinations/destination-sdk/docs-framework/self-service-template.md).

```
1. Next, you are shown a wizard which assists you in configuring the Salesforce CRM destination. Create the new destination by selecting **[!UICONTROL Configure new destination]** and fill in the details below and select **[!UICONTROL Connect to destination]**.

```

vfalon commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+![Account Info](../../assets/catalog/crm/salesforce/account-info.png)
+
+*  **[!UICONTROL Password]**: Your Salesforce account password.
+*  **[!UICONTROL Client ID]**: Consumer Key generated during the [prerequisites](#prerequisites) step.
```
For client ID and client Secret, instead of linking to the prerequisites section, let's link to the Salesforce documentation that shows where to find them.

mohan-h replied:
Have reworded and removed the reference to prerequisites.

vfalon commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+
+*  **[!UICONTROL Name]**: A name by which you will recognize this destination in the future.
+*  **[!UICONTROL Description]**: A description that will help you identify this destination in the future.
+*  **[!UICONTROL Custom Domain]**: Your Salesforce domain as shown within [prerequisites](#prerequisites).
```
Line 109: there is no mention of domain in the prereqs. Please link to the Salesforce docs that show how to obtain the domain instead.

mohan-h replied:
Added a reference to the domain in prerequisites, also reworded and removed the reference to prerequisites.

vfalon commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+Read [Activate profiles and segments to streaming segment export destinations](../../ui/activate/activate-segment-streaming-destinations.md) for instructions on activating audience segments to this destination.
+
+The list of attribute mappings that can be set up for the [Salesforce REST API](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_composite_upsert_example.htm?q=contacts) is given below.
+
```
This table of identities should be moved to the Supported identities section. Please update that table according to what Salesforce supports, as mentioned in the template: https://experienceleague.adobe.com/docs/experience-platform/destinations/destination-sdk/document-destination/self-service-template.html?lang=en#supported-identities

mohan-h replied:
These are not linked to the *Supported identities*, its for [Map attributes and identities](https://experienceleague.adobe.com/docs/experience-platform/destinations/ui/activate/activate-segment-streaming-destinations.html?lang=en#mapping)

mohan-h replied:
[Vlad Falon](https://app.slack.com/team/U03BEN7G3RA)  [7:58 PM](https://launch-developers.slack.com/archives/D03G7C72HJ5/p1653402533002859)
[@Mohan Hariharan](https://launch-developers.slack.com/team/U030SRCTJEM) regarding the last comment, those are the supported identities. Supported identities describes what type of identifiers you can map in that step. Whatever Salesforce supports needs to go in the supported identities table.
 
[7:59](https://launch-developers.slack.com/archives/D03G7C72HJ5/p1653402552982979)
the activation section should be the same for all destinations, just like the template shows

vfalon reviewed (changes requested):
Thanks for opening the PR! Please see my comments before we can move forward with the publishing process.

mohan-h committed:
Updates against review comments

Updates against review comments https://github.com/AdobeDocs/experience-platform.en/pull/60#pullrequestreview-982819246

mohan-h commented:
Thanks @vfalon have incorporated all changes except for the last one.  Marked a resolved / commented against the reviews.

mohan-h committed:
Updates against review comments

Updates against review comments

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,142 @@
+
+[Salesforce CRM](https://www.salesforce.com/) is a popular Customer Relationship Management (CRM) platform.
+
+This [!DNL Adobe Experience Platform] [destination](/help/destinations/home.md) leverages the [Salesforce REST API](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_composite_upsert_example.htm?q=contacts), which allows you to add Accounts and create Contacts after activating them within a new Salesforce segment for your business needs.
```
what is the reason for using uppercase Accounts and Contacts?

mohan-h replied:
Was meant to refer to Salesforce entities but you are correct its not related to AEP have changed to lowercase.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,142 @@
+
+This [!DNL Adobe Experience Platform] [destination](/help/destinations/home.md) leverages the [Salesforce REST API](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_composite_upsert_example.htm?q=contacts), which allows you to add Accounts and create Contacts after activating them within a new Salesforce segment for your business needs.
+
+Salesforce CRM uses [OAuth 2 with Password Grant](https://experienceleague.adobe.com/docs/experience-platform/destinations/destination-sdk/functionality/authentication/oauth2-authentication.html?lang=en) as an authentication mechanism to communicate with the Salesforce REST API.
```
Can we rephrase to "the Salesforce CRM destination uses OAuth 2 with Password Grant as an authentication mechanism to communicate with the Salesforce REST API. Instructions are further below (link to line 83).

I suggest we remove the link you provided (https://experienceleague.adobe.com/docs/experience-platform/destinations/destination-sdk/functionality/authentication/oauth2-authentication.html?lang=en) since this link is useful for partners implementing the destination (e.g. Infosys in this case),rather than Platform users.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,142 @@
+
+To help you better understand how and when you should use the Salesforce CRM destination, here are sample use cases that Adobe Experience Platform customers can solve by using this destination.
+
+### Use case
```
Let's spell out in the title the use case described below instead of just leaving `Use case` here.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,142 @@
+
+Before activating data to the Salesforce CRM destination, you must have a [schema](https://experienceleague.adobe.com/docs/experience-platform/xdm/schema/composition.html), a [dataset](https://experienceleague.adobe.com/docs/platform-learn/tutorials/data-ingestion/create-datasets-and-ingest-data.html?lang=en), and [segments](https://experienceleague.adobe.com/docs/platform-learn/tutorials/segment ...
+
+The following items are required before you start configuring the destination.
```
Let's add some more context: 

The following items are required in Salesforce before you start configuring the destination.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,142 @@
+1. Note down your [Salesforce domain](https://help.salesforce.com/s/articleView?id=sf.domain_name_setting_login_policy.htm&type=5).
+1. Create a [connected app](https://help.salesforce.com/s/articleView?id=sf.connected_app_create.htm&language=en_US&r=https%3A%2F%2Fhelp.salesforce.com%2F&type=5), enable its [OAuth Settings](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US) and once created successfully note down its *Consumer Key* and *Consumer Secret*. Refer to the [Sa ...
+1. Select the following [scopes](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US) for the connected app in the Salesforce user interface:
+    - chatter_api
```
let's use `*` instead of `-` for all the scopes. Our markdown validator won't accept lists with `*` and `-` in the same page, and we already used `*` further above.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,142 @@
+| postalCode | Mailing Postal Code ||
+| city | Mailing City ||
+| lastName | Last Name ||
+| email |Salesforce Email ID|Note that *Salesforce CRM* does not support hashed email addresses, hence only plain text data without transformation is sent to the destination.|
```
Apologies for the back-and-forth on this one, but we need to make a change. 

`email` is the only identity in this table, all the other items are **attributes**. 

Regarding the other attributes, are these the target fields available to users? (similar to this example from another destination)

![Screen Shot 2022-06-02 at 11 45 35 AM](https://user-images.githubusercontent.com/19704189/171669211-2ee95718-4bff-4f8e-8167-c89cc67257d0.png)

mohan-h replied:
Many thanks for the clarification. Was a tad concerned about this one.
Yes these fields are available to users. There would be more possible but it depends on what requires to be pushed. 
Referring to guidance from Mixpanel (Source) https://github.com/AdobeDocs/experience-platform.en/pull/64#discussion_r887387266 have added backticks.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,142 @@
+*  **[!UICONTROL Client Secret]**: Your Salesforce connected app Consumer Secret.
+*  **[!UICONTROL Username]**: Your Salesforce account username.
+
+1. If the details provided are valid, the UI displays a **Connected** status with a green check mark, you can then proceed to the next step.
```
We can remove the `1. ` from the start of the line here.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,142 @@
+
+Read [Activate profiles and segments to streaming segment export destinations](../../ui/activate/activate-segment-streaming-destinations.md) for instructions on activating audience segments to this destination.
+
+## Exported data / Validate data export {#exported-data}
```
Let's just leave Validate data export {#validate-data-export} here

vgiurgiu reviewed (commented):
@mohan-h  - I left some comments. The page looks great, but please address these comments before we can publish.

mohan-h committed:
Merge branch 'AdobeDocs:main' into salesforce-crm-dest

mohan-h committed:
Updates from Review guidance

Incorporated review guidance from https://github.com/AdobeDocs/experience-platform.en/pull/60#pullrequestreview-993733758

mohan-h committed:
Image updated to show only domain prefix

Image updated to show only domain prefix

mohan-h committed:
Added custom domain example.

Added custom domain example.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,155 @@
+
+This [!DNL Adobe Experience Platform] [destination](/help/destinations/home.md) leverages the [Salesforce REST API](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_composite_upsert_example.htm?q=contacts), which allows you to add accounts and create contacts after activating them within a new Salesforce segment for your business needs.
+
+Salesforce CRM uses OAuth 2 with Password Grant as an authentication mechanism to communicate with the Salesforce REST API. Instructions are further [below](#authenticate).
```
Minor comment: 

Instructions to authenticate to your Salesforce CRM instance are further below, in the [Authenticate to destination](#authenticate) section.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,155 @@
+
+Before activating data to the Salesforce CRM destination, you must have a [schema](https://experienceleague.adobe.com/docs/experience-platform/xdm/schema/composition.html), a [dataset](https://experienceleague.adobe.com/docs/platform-learn/tutorials/data-ingestion/create-datasets-and-ingest-data.html?lang=en), and [segments](https://experienceleague.adobe.com/docs/platform-learn/tutorials/segment ...
+
+In order to access your Salesforce account on Platform.
```
Note the following prerequisites in Salesforce, in order to export data from Platform to your Salesforce account:

vgiurgiu replied:
Note the following prerequisites in Salesforce, in order to export data from Platform to your Salesforce account:

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,155 @@
+1. You need to have a Salesforce account.
+    * Go to the Salesforce [trial](https://www.salesforce.com/in/form/signup/freetrial-sales/) page to register and create a Salesforce account, if you do not have one already.
+1. Next, You need to configure a [connected app](https://help.salesforce.com/s/articleView?id=sf.connected_app_create.htm&language=en_US&r=https%3A%2F%2Fhelp.salesforce.com%2F&type=5) within your Salesforce account, if you do not have one already. 
+    * Within the connected app ensure [OAuth Settings](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US) is enabled.
```
Minor: ensure _that_

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,155 @@
+    * Go to the Salesforce [trial](https://www.salesforce.com/in/form/signup/freetrial-sales/) page to register and create a Salesforce account, if you do not have one already.
+1. Next, You need to configure a [connected app](https://help.salesforce.com/s/articleView?id=sf.connected_app_create.htm&language=en_US&r=https%3A%2F%2Fhelp.salesforce.com%2F&type=5) within your Salesforce account, if you do not have one already. 
+    * Within the connected app ensure [OAuth Settings](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US) is enabled.
+    * Also ensure the [scopes](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US) mentioned below are selected.
```
Minor: ensure _that_

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,155 @@
+        * web
+        * refresh_token
+        * offline_access
+1. Note down the items below before you Authenticate to the Salesforce CRM destination.
```
Minor: lowercase `a` in Authenticate

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,155 @@
+
+Read [Activate profiles and segments to streaming segment export destinations](../../ui/activate/activate-segment-streaming-destinations.md) for instructions on activating audience segments to this destination.
+
+The list of attribute mappings that can be set up for the [Salesforce REST API](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_composite_upsert_example.htm?q=contacts) is given below.
```
Great table!

vgiurgiu reviewed (commented):
Very good updates! I left just a few comments, and they're pretty minor. 

One thing to keep in mind though is if the Salesforce API does not support upsert operations. If this has an effect on customers' integrations and how profiles are added and updated in Salesforce from Platform, we should call it out in the documentation.

mohan-h committed:
Merge branch 'AdobeDocs:main' into salesforce-crm-dest

mohan-h committed:
Merge branch 'AdobeDocs:main' into salesforce-crm-dest

mohan-h committed:
Merge branch 'AdobeDocs:main' into salesforce-crm-dest

mohan-h committed:
Updates post review

Updates post [review](https://github.com/AdobeDocs/experience-platform.en/pull/60#pullrequestreview-996830009)

mohan-h committed:
Delete salesforce.pdf

Delete salesforce.pdf

juliette-massardier committed:
Update license-usage.md

Hello, as discussed with the Adobe Journey Optimizer product team, we should add the AJO licenses listed here and the link to the AJO product description. Thanks! Juliette (AJO doc team)

JCON3DEV committed:
Merge pull request #80 from juliette-massardier/patch-1

Update AJO license-usage.md

mohan-h committed:
Merge branch 'AdobeDocs:main' into salesforce-crm-dest

mohan-h committed:
Changed to update only and uses custom field

Connector implementation has changed to update only (no create, or upsert) and requires a custom field in Salesforce.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+
+Note the following details that are specific to the [!DNL Salesforce CRM] destination:
+
+* [!DNL Adobe Experience Platform] segments are exported to [!DNL Salesforce CRM] using a custom `AdobeExperiencePlatformSegments` field. Instructions are further below, in the [Prerequisites](#prerequisites) section.
```
Let's be a bit more specific here. Instead of "Instructions are further below, in the [Prerequisites](#prerequisites) section.", let's call out instructions to do what exactly.

mohan-h replied:
@vgiurgiu Wouldn't that create content duplication as I'd be adding the below

```
* Create `AdobeExperiencePlatformSegments` as a custom field to allow Experience Platform to update your profiles within Salesforce CRM. Refer to the Salesforce documentation to [create custom fields](https://help.salesforce.com/s/articleView?id=sf.adding_fields.htm&type=5) if you need additional guidance.

    | Field | Data Type |
    |---|---|
    | `AdobeExperiencePlatformSegments` | `Text Area Long` |
```
And then within prerequisites I would have to add a preamble *If you have not already done so,*

```
1. If you have not already done so, create `AdobeExperiencePlatformSegments` as a custom field to allow Experience Platform to update your profiles within Salesforce CRM. Refer to the Salesforce documentation to [create custom fields](https://help.salesforce.com/s/articleView?id=sf.adding_fields.htm&type=5) if you need additional guidance.

    | Field | Data Type |
    |---|---|
    | `AdobeExperiencePlatformSegments` | `Text Area Long` |

```

Would rewording it as below work ?

* [!DNL Adobe Experience Platform] segments are exported to [!DNL Salesforce CRM] using a custom field. The details for which are further below, in the [Prerequisites](#prerequisites) section.

vgiurgiu replied:
Thanks for the rewording suggestion. Just a tiny tweak and I think we got it. How about the version below?: 

[!DNL Adobe Experience Platform] segments are exported to [!DNL Salesforce CRM] using a custom `AdobeExperiencePlatformSegments` field. You need to create this custom field in [!DNL Salesforce] first, before you export data from Experience Platform. Instructions are further below, in the [Prerequisites](#prerequisites) section.

mohan-h replied:
Sure, will update to the same.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+
+1. You need to have a Salesforce account.
+    * Go to the Salesforce [trial](https://www.salesforce.com/in/form/signup/freetrial-sales/) page to register and create a Salesforce account, if you do not have one already.
+1. Next, You need to configure a [connected app](https://help.salesforce.com/s/articleView?id=sf.connected_app_create.htm&language=en_US&r=https%3A%2F%2Fhelp.salesforce.com%2F&type=5) within your Salesforce account, if you do not have one already. 
```
minor typo: let's use lowercase you

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+    * Go to the Salesforce [trial](https://www.salesforce.com/in/form/signup/freetrial-sales/) page to register and create a Salesforce account, if you do not have one already.
+1. Next, You need to configure a [connected app](https://help.salesforce.com/s/articleView?id=sf.connected_app_create.htm&language=en_US&r=https%3A%2F%2Fhelp.salesforce.com%2F&type=5) within your Salesforce account, if you do not have one already. 
+    * Within the connected app ensure that [OAuth Settings](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US) is enabled.
+    * Also ensure that [scopes](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US) mentioned below are selected.
```
minor: ensure that _the_ scopes

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+1. Next, You need to configure a [connected app](https://help.salesforce.com/s/articleView?id=sf.connected_app_create.htm&language=en_US&r=https%3A%2F%2Fhelp.salesforce.com%2F&type=5) within your Salesforce account, if you do not have one already. 
+    * Within the connected app ensure that [OAuth Settings](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US) is enabled.
+    * Also ensure that [scopes](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US) mentioned below are selected.
+        * chatter_api
```
let's please wrap the scopes in `` symbols like this: `chatter_api`

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+        * web
+        * refresh_token
+        * offline_access
+1. Create the custom field which will allow Experience Platform to update your profiles within Salesforce CRM. Refer to the Salesforce documentation to [Create Custom Fields](https://help.salesforce.com/s/articleView?id=sf.adding_fields.htm&type=5) if you need additional guidance.
```
let's use lowercase create custom fields

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+
+## Mapping Example {#mapping-example}
+
+In the select target window, When selecting the target field choose the **Select custom attributes** category and add the mappings' desired.
```
please consider rewriting this sentence as it is very difficult to follow and contains several errors like: 

* uppercase When after the comma
* unnecessary apostrophe after mappings.

My suggestion is to break it down into several sentences and add a screenshot for context. Thanks!

mohan-h replied:
Updated, couldn't manage to explain with just one screenshot. Had to elaborate the entire section.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+
+In the select target window, When selecting the target field choose the **Select custom attributes** category and add the mappings' desired.
+
+For instance, your XDM profile schema and your [!DNL Salesforce CRM] instance could have the attribute mapping as below:
```
Minor rewording suggestion: 

For instance, you could add the following mapping between your XDM profile schema and your [!DNL Salesforce CRM] instance:

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+1. Monitor the segment summary and ensure that the count of profiles corresponds to the count created within the segment.
+![Segment](../../assets/catalog/crm/salesforce/segment.png)
+
+1. Login to the Salesforce website then navigate to the **[!DNL Apps]** > **[!DNL Contacts]** page and check if the profiles from the segment have been added.
```
Minor: missing comma after website

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+1. Login to the Salesforce website then navigate to the **[!DNL Apps]** > **[!DNL Contacts]** page and check if the profiles from the segment have been added.
+![Salesforce Contacts](../../assets/catalog/crm/salesforce/contacts.png)
+
+1. Click a contact and check if the attribute is updated. You will notice the segment id's which were used to update against the `AdobeExperiencePlatformSegments` custom field.
```
Minor rewording suggestion: 

You will notice the segment IDs from Experience Platform in the `AdobeExperiencePlatformSegments` custom field.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,161 @@
+![Salesforce Contacts](../../assets/catalog/crm/salesforce/contacts.png)
+
+1. Click a contact and check if the attribute is updated. You will notice the segment id's which were used to update against the `AdobeExperiencePlatformSegments` custom field.
+![Salesforce Contacts](../../assets/catalog/crm/salesforce/contact-info.png)
```
This image has a lot of unnecessary whitespace at the bottom. Please edit to remove the whitespace.

vgiurgiu reviewed (commented):
@mohan-h - thanks for the update. Please address my comments before we can publish.

mohan-h committed:
Updates against review

Updates against review https://github.com/AdobeDocs/experience-platform.en/pull/60#pullrequestreview-1020682431

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,180 @@
+
+Read [Activate profiles and segments to streaming segment export destinations](../../ui/activate/activate-segment-streaming-destinations.md) for instructions on activating audience segments to this destination.
+
+## Mapping Example {#mapping-example}
```
Considering all the extra information you added, I think that we can now update the section title to:

```# Mapping considerations and example {#mapping-considerations-example}```

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,180 @@
+
+## Mapping Example {#mapping-example}
+
+* To correctly send your audience data from Adobe Experience Platform to the Salesforce CRM destination, you need to go through the field mapping step.
```
There is no need for the three bullet points. We could just collapse the three sentences into one paragraph.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,180 @@
+
+* To correctly send your audience data from Adobe Experience Platform to the Salesforce CRM destination, you need to go through the field mapping step.
+
+* Mapping consists of creating a link between your Experience Data Model (XDM) schema fields in your Platform account, and their corresponding equivalents from the target destination.
```
minor: no comma needed after account

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,180 @@
+
+* To correctly map your XDM fields to the Salesforce CRM destination fields, follow these steps:
+
+    1. In the Mapping step, click **[!UICONTROL Add new mapping]**, you will see a new mapping row on the screen.
```
there is no need for indentation in the numbered list, we can bring back all the sentences to the start of the row.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,180 @@
+
+## Use cases {#use-cases}
+
+As a marketer, I can deliver personalized experiences to my users, based on attributes from their Adobe Experience Platform profiles, build segments from my offline data, and send these segments to Salesforce CRM, to display in my users’ feeds, as soon as segments and profiles are updated in Adobe Experience Platform.
```
Sorry, I missed this earlier. Can we please rephrase from 1st person to 3rd person? Also, as a general rule, let's please try to break up long sentences into smaller, separate sentences. After three or four commas, a sentence becomes difficult to follow. See an improved example below:

As a marketer, you can deliver personalized experiences to your users, based on attributes from their Adobe Experience Platform profiles. You can  build segments from your offline data and send these segments to Salesforce CRM, to display in your users’ feeds as soon as segments and profiles are updated in Experience Platform.

mohan-h replied:
Apologies, had forgotten the 1st person to 3rd person. Comma's gotcha.

vgiurgiu reviewed (commented):
Getting there. Just a few more comments to address. As always, thanks for the updates and the collaboration.

mohan-h committed:
Updates against review

Updates against review https://github.com/AdobeDocs/experience-platform.en/pull/60#pullrequestreview-1021788012

mohan-h committed:
Merge branch 'AdobeDocs:main' into salesforce-crm-dest

mohan-h committed:
Merge branch 'salesforce-crm-dest' of https://github.com/mohan-h/experience-platform.en into salesforce-crm-dest

mohan-h committed:
Segment status updated into CRM custom field

Segment status is now updated into salesforce CRM custom field

mohan-h committed:
Merge branch 'AdobeDocs:main' into salesforce-crm-dest

mohan-h committed:
Merge branch 'AdobeDocs:main' into salesforce-crm-dest

mohan-h committed:
avoid whitespace in name of custom field Salesforce

Added note to avoid whitespace in the name of the custom field attributes within Salesforce

mohan-h committed:
Incorporate self-service-template updates

Incorporate self-service-template updates

mohan-h committed:
Content correction

mohan-h committed:
Added errors & troubleshooting, minor cleanup

Added errors & troubleshooting; Removed the superflous content from destination specifics merging into overview.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,202 @@
+
+|Target Identity|Description|Considerations|
+|---|---|---|
+| SalesforceId |Custom Salesforce CRM identifier that supports mapping of any identity.|Mandatory, You can send any [identity](../../../identity-service/namespaces.md) to the [!DNL Salesforce CRM] destination, as long as you map it to the `SalesforceId`. |
```
Minor, please update comma to full stop below: 

"Mandatory. You"

mohan-h replied:
Moved the text to the Description column.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,202 @@
+
+## Prerequisites {#prerequisites}
+
+Before activating data to the Salesforce CRM destination, you must have a [schema](https://experienceleague.adobe.com/docs/experience-platform/xdm/schema/composition.html), a [dataset](https://experienceleague.adobe.com/docs/platform-learn/tutorials/data-ingestion/create-datasets-and-ingest-data.html?lang=en), and [segments](https://experienceleague.adobe.com/docs/platform-learn/tutorials/segment ...
```
Regarding the prerequisites section, I am concerned that the numbered list further below might not render. I suggest adding some subheadings instead, like this: 

```

### Prerequisites in Experience Platform {#prerequisites-in-experience-platform}

Before activating data to the Salesforce CRM destination, you must have a [schema](https://experienceleague.adobe.com/docs/experience-platform/xdm/schema/composition.html), a [dataset](https://experienceleague.adobe.com/docs/platform-learn/tutorials/data-ingestion/create-datasets-and-ingest-data.html?lang=en), and [segments](https://experienceleague.adobe.com/docs/platform-learn/tutorials/segments/create-segments.html?lang=en) created in [!DNL Experience Platform].

### Prerequisites in Salesforce CRM

Note the following prerequisites in Salesforce, in order to export data from Platform to your Salesforce account:

#### You need to have a Salesforce account

Go to the Salesforce [trial](https://www.salesforce.com/in/form/signup/freetrial-sales/) page to register and create a Salesforce account, if you do not have one already.

#### Configure a connected app

Next, you need to configure a [connected app](https://help.salesforce.com/s/articleView?id=sf.connected_app_create.htm&language=en_US&r=https%3A%2F%2Fhelp.salesforce.com%2F&type=5) within your Salesforce account, if you do not have one already. 

Within the connected app, ensure that [OAuth Settings](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US) is enabled.

Also ensure that the [scopes](https://help.salesforce.com/s/articleView?id=connected_app_create_api_integration.htm&type=5&language=en_US) mentioned below are selected.
        * ``chatter_api``
        * ``lightning``
        * ``visualforce``
        * ``content``
        * ``openid``
        * ``full``
        * ``api``
        * ``web``
        * ``refresh_token``
        * ``offline_access``

#### Create custom field

Create a custom field of type `Text Area Long`, which Experience Platform will use to update the segment status within Salesforce CRM. 
Refer to the Salesforce documentation to [create custom fields](https://help.salesforce.com/s/articleView?id=sf.adding_fields.htm&type=5) if you need additional guidance. 

> [!IMPORTANT]
>
> Ensure there are no whitespace characters in the name, instead use the underscore i.e. the `_` character as a separator.

> [!NOTE]
>
> * Objects in Salesforce are restricted to 25 External fields, see [Custom Field Attributes](https://help.salesforce.com/s/articleView?id=sf.custom_field_attributes.htm&type=5).
> * This restriction implies you will only be able to have at a maximum 25 Experience Platform segment memberships active at any time. 
> * If you have reached this limit within Salesforce, you will need to remove the custom attribute from Salesforce that were used to store the segment status against older segments within Experience Platform before a new mappingId can be used.

Refer to the Experience Platform documentation for [Segment Membership Details schema field group](https://experienceleague.adobe.com/docs/experience-platform/xdm/field-groups/profile/segmentation.html?lang=en) if you need guidance on segment statuses.

#### Gather Salesforce credentials

Note down the items below before you authenticate to the Salesforce CRM destination:

    * Note down your [Salesforce domain prefix](https://help.salesforce.com/s/articleView?id=sf.domain_name_setting_login_policy.htm&type=5). For example if your domain is *`d5i000000isb4eak-dev-ed`.my.salesforce.com*, you need the highlighted value.
    * Copy the *Consumer Key* and *Consumer Secret* from the connected app. Refer to the [Salesforce documentation](https://help.salesforce.com/s/articleView?id=sf.connected_app_rotate_consumer_details.htm&type=5) if you need additional guidance.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,202 @@
+
+To correctly send your audience data from Adobe Experience Platform to the Salesforce CRM destination, you need to go through the field mapping step. Mapping consists of creating a link between your Experience Data Model (XDM) schema fields in your Platform account and their corresponding equivalents from the target destination. To correctly map your XDM fields to the Salesforce CRM destination f ...
+
+1. In the Mapping step, click **[!UICONTROL Add new mapping]**, you will see a new mapping row on the screen.
```
The mapping considerations and example section is such a helpful and well written section. Well done!

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,202 @@
+
+### Schedule segment export and example {#schedule-segment-export-example}
+
+* When performing the [Schedule segment export](../../ui/activate/activate-segment-streaming-destinations.html?lang=en#scheduling) step you will need to manually map Platform segments to the custom field attribute in Salesforce.
```
Minor wording update. Currently, it reads like this step is somehow optional. Suggestion to reword to: 

In the [Schedule segment export](../../ui/activate/activate-segment-streaming-destinations.html?lang=en#scheduling) step, you must to manually map Platform segments to the custom field attribute in Salesforce.

vgiurgiu replied:
I don't think that using bullet points here is appropriate. Generally, bullet points should be used when we are listing something. In this case, I suggest removing all three bullet points from this section.

vgiurgiu commented on help/destinations/catalog/crm/salesforce.md:
```diff
@@ -0,0 +1,202 @@
+
+## Errors and troubleshooting {#errors-and-troubleshooting}
+
+When checking a dataflow run, if you obtain the below error message; Check that the Mapping ID you provided against your Platform segment is valid and exists within [!DNL Salesforce CRM].
```
Minor comment here. Considering that maybe we'll want to add some more errors in the future, we could have a subheading for each error. So for the existing one, we would have: 

```
### Unknown errors encountered while pushing events to destination

When checking a dataflow run, if you see the error message below, verify that the **Mapping ID** that you provided in [!DNL Salesforce CRM] for your Platform segment is valid and exists within [!DNL Salesforce CRM].
```

vgiurgiu reviewed (commented):
@mohan-h - I left a few comments based on your latest updates. Once these are resolved, we should be ready to publish this destination.

mohan-h committed:
Updates against review

Updates against review https://github.com/AdobeDocs/experience-platform.en/pull/60#pullrequestreview-1039368976

mohan-h committed:
Minor format correction

mohan-h committed:
Updates against review

mohan-h committed:
Merge branch 'AdobeDocs:main' into salesforce-crm-dest

mohan-h committed:
Added limits section

vgiurgiu commented:
The documentation for the new Salesforce CRM destination is live at:

https://experienceleague.adobe.com/docs/experience-platform/destinations/catalog/crm/salesforce.html?lang=en
